src/content/blog/astro-480.mdx

```diff
@@ -0,0 +1,177 @@
+
+This release includes the first preview for a long awaited feature: Astro actions! Actions make it easy to define and call backend functions with full type-safety from your client code.
+
+To use this feature, [add a server build output](https://docs.astro.build/en/basics/rendering-modes/#on-demand-rendered) and enable the `experimental.actions` option in your Astro config:
+
+```js
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  output: "hybrid", // or 'server'
+  experimental: {
+    actions: true,
+  },
+});
+```
+
+Then, define actions in a `src/actions/index.ts` file using `defineAction()`. You can define actions that accept JSON or form requests, and the `handler` function will be called with your type-safe input.
+
+Plus, no more casting `formData.get()` results. Astro will automatically parse form requests to objects using your Zod schema. Here's an example of a newsletter signup action:
+
+```ts
+// src/actions/index.ts
+import { defineAction, z } from "astro:actions";
+
+export const server = {
+  newsletter: defineAction({
+    accept: "form",
+    input: z.object({
+      email: z.string().email(),
+      receivePromo: z.boolean(),
+    }),
+    handler: async ({ email, receivePromo }) => {
+      // call a mailing service, or store to a database
+      return { success: true };
+    },
+  }),
+};
+```
+
+You can call actions from any client component using the `actions` object from `astro:actions`. You can pass a type-safe object when using JSON, or a `FormData` object when using `accept: 'form'`. You can also use `getActionProps()` for progressive enhancement:
+
+```jsx
+// src/components/newsletter.tsx
+import { actions, getActionProps } from "astro:actions";
+
+export function Newsletter() {
+  return (
+    <form
+      method="POST"
+      onSubmit={async (e) => {
+        e.preventDefault();
+        const formData = new FormData(e.target);
+        const result = await actions.newsletter(formData);
+      }}
+    >
+      <input {...getActionProps(actions.newsletter)} />
+      <label for="email">Email</label>
+      <input name="email" type="email" id="email" />
+
+      <label for="receivePromo">Receive promotional emails</label>
+      <input name="receivePromo" type="checkbox" id="receivePromo" checked />
+
+      <button type="submit">Sign Up</button>
+    </form>
+  );
+}
+```
+
+For more information on Astro actions, visit [the experimental actions docs](https://docs.astro.build/en/reference/configuration-reference/#experimentalactions).
+
+## Experimental: Request Rewriting
+
+Astro 4.8 introduces experimental support for Request rewriting. Rewriting is a routing feature that allows you to show the content of a different page from the current URL. This can be useful for showing different content based on the user's location, device, or other conditions without needing multiple different URLs. This feature is sometimes also called "rerouting" in other frameworks.
```

Comment:
```suggestion
Astro 4.8 introduces experimental support for `Request` rewriting. Rewriting is a routing feature that allows you to show the content of a different page from the current URL. This can be useful for customizing content based on the user's location, device, or other conditions without needing multiple, different URLs. This feature is sometimes also called "rerouting" in other frameworks.
```
- If talking about the standard `Request` object, we'd typically inline code that thing (Is that the case here?)
- Maybe we only need one of "multiple" and "different"? If you do want to emphasize both words, then a comma goes here.
- suggested a change to "customizing content" just to avoid two "different" one right after the other.

Reply:
I would argue that it should be `request rewriting` here, because the concept isn't really specific to `Request`!